### DISCUSSION TITLE
Flutter Web Supabase Reset Password Flow

### DISCUSSION THREAD

**Discussion by JulianSwales** [supabase/supabase#14060] [Questions] [answered]
I am attempting to get the password reset flow from Supabase working for a Flutter Web App.. 
I have the password reset request working.. I get the email with the correct reset values and redirect URL.. 
However it fails when clicking the link.. I have this listener running: 
```
  final authSubscription = supabase.auth.onAuthStateChange.listen((data) {
    final AuthChangeEvent event = data.event;
    final Session? session = data.session;
    debugPrint('Auth State Change $event');
  });
```
from which I see AuthChangeEvent.signedIn when I do a normal signIn and AuthChangeEvent.SignedOut when I signet.. But when clicking the password reset link sent in the email I get nothing.. Soo, what am I doing incorrectly? 

https://zzzzzzzzz.supabase.co/auth/v1/verify?token=1111111111&type=recovery&redirect_to=https://xxx-yyy.com/profile/changePassword

The above is the URL I get in the email with the instance, token and host name obfuscated.. 

As seen below, I even have a DeepLink observer working when app starts.. 
Flutter Web Bootstrap: Programmatic 
***** Supabase init completed Instance of 'minified:ael'
Got object store box in database supabase_authentication. 
***** SupabaseDeepLinkingMixin startAuthObserver 
Got object store box in database authuser.
Got object store box in database config.

I'm running supabase_flutter: ^1.8.1

Any suggestions on how I can solve this? Thanks, Julian..

**Comment by dshukertjr** [comment]
Configuring URL strategy might fix it. 

https://docs.flutter.dev/ui/navigation/url-strategies#configuring-the-url-strategy

**Comment by JulianSwales** [comment, marked as answer]
Turns out the problem was I had a value in the authCallBackUrlHostname key of the initialize statement. Because of that, the listener doesn't recognize the response and ignores it.. Comment it out, as below, and voila I start to see the passwordRecover events. 
```
await Supabase.initialize(
    url: getIt<EnvironmentDetails>().getSupabaseUrl(),
    anonKey: getIt<EnvironmentDetails>().getSupabaseKey(),
    //authCallbackUrlHostname: 'login-callback',
    debug: true,
  );```

Thanks. Julian..
